### PR TITLE
pr_checker: Added PowerShell/Python/Bash checks

### PR DESCRIPTION
**Pull request by ader1990**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -30,12 +30,40 @@ jobs:
...
+    cancelTimeoutInMinutes: 12
+    strategy:
+      parallel: 1
+    pool:
+      vmImage: 'Ubuntu 16.04'
+    steps:
+    - task: UsePythonVersion@0
+    - powershell:
+        python -m pip install --upgrade flake8;
+        flake8 --select=W291,F841,F401;
```
[the remark this answers is not in the document]
Comment (ader1990) [reply azure-pipelines.yml]: flake8 finds also logic/code errors and flake8 is based on pep8.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -30,12 +30,40 @@ jobs:
...
+  - job: 'BashCodeCheck_ShellCheck'
+    timeoutInMinutes: 10
+    cancelTimeoutInMinutes: 12
+    strategy:
+      parallel: 1
+    pool:
+      vmImage: 'Ubuntu 16.04'
+    steps:
+    - script:
+        sudo apt update > /dev/null && sudo apt install -y shellcheck > /dev/null && shellcheck -e SC2003,SC2129,SC2155,SC2145,SC2144,SC2125,SC2016,SC2009,SC2143,SC2010,SC2126,SC2124,SC2154,SC2068,SC2029,SC2128,SC1017,SC2181,SC1091,SC2086,SC2006,SC2046,SC2053,SC1083,SC2076,SC2004,SC2063,SC2002,SC2001 ./Testscripts/Linux/*.sh
```
Comment (ader1990) [inline azure-pipelines.yml]: shell check does not have a way to just include the wanted errors, so we have to exclude all the ones we do not want.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -30,12 +30,40 @@ jobs:
     - powershell: Install-Module -Name PSScriptAnalyzer -Scope CurrentUser -Force -SkipPublisherCheck -AllowClobber
     - powershell:
         $rules = @(
-                 "PSUseDeclaredVarsMoreThanAssignments", "PSAvoidTrailingWhitespace"
+                 "PSUseDeclaredVarsMoreThanAssignments", "PSAvoidTrailingWhitespace", "PSAvoidUsingEmptyCatchBlock"
```
[the remark this answers is not in the document]
Comment (ader1990) [reply azure-pipelines.yml]: if you refer to PSUseDeclaredVarsMoreThanAssignments is already added at the begining

**Comment by juhlee-microsoft** [comment]
@ader1990 3 PCChecker job failed. Will you resolve all?

**Comment by lizzha** [comment]
@ader1990 will you fix the errors for the new check rules?

**Comment by ader1990** [comment]
please keep this PR opened until I also fix the errors

**Comment by ader1990** [comment]
@juhlee-microsoft @lizzha please merge this PR, as I have fixed the powershell and python warnings.
I left the bash checker warnings fix on a subsequent pull request, as it requires a high amount of work and also testing.

**Pull request merged**